src/client.js

```diff
@@ -97,8 +91,15 @@ class StitchClient {
    * @param {Object} [options] additional authentication options (user data)
    * @returns {Promise}
    */
-  authenticate(providerType, options) {
```

Comment:
We need to update the JS doc for this function, a few reasons:
- It's not OAuth specific at all, since you could actually pass in any `providerType`, which we're now doing explicitly in other client functions.
- The additional change in behavior (reusing existing auth if present) is worth mentioning
- A note on the `@returns {Promise}` line that says that the promise resolves to the userId (string type) would be helpful.